concent_api/concent_api/tests/test_shutdown_mode.py

```diff
@@ -0,0 +1,329 @@
+            expected_nested_messages = {'task_to_compute', 'report_computed_task', 'ack_report_computed_task'},
+        )
+        self._test_last_stored_messages(
+            expected_messages= [
+                message.concents.AckReportComputedTask,
+            ],
+            task_id         = '1',
+            subtask_id      = '8',
+            timestamp       = "2017-12-01 11:00:05"
+        )
+        self._test_undelivered_pending_responses(
+            subtask_id                         = '8',
+            client_public_key                  = self._get_encoded_provider_public_key(),
+            expected_pending_responses_receive = [
+                PendingResponse.ResponseType.ForceReportComputedTaskResponse,
+            ]
+        )
+        self.assertEqual(len(mail.outbox), len(settings.ADMINS))
+
+    def test_in_shutdown_mode_concent_should_not_accept_payment_requests(self):
+        """
+        Tests if in shutdown mode Concent will not accept new payment requests.
+        """
+
+        subtask_results_accepted_list = [
+            self._get_deserialized_subtask_results_accepted(
+                timestamp       = "2018-02-05 10:00:15",
+                payment_ts      = "2018-02-05 12:00:16",
+                task_to_compute = self._get_deserialized_task_to_compute(
+                    timestamp                       = "2018-02-05 10:00:00",
+                    deadline                        = "2018-02-05 10:00:10",
+                    task_id                         = '2',
+                    requestor_public_key            = self._get_encoded_requestor_public_key(),
+                    requestor_ethereum_public_key   = self._get_encoded_requestor_ethereum_public_key(),
+                )
+            )
+        ]
+        serialized_force_payment = self._get_serialized_force_payment(
+            timestamp                     = "2018-02-05 10:00:15",
+            subtask_results_accepted_list = subtask_results_accepted_list
+        )
+
+        with freeze_time("2018-02-05 10:00:25"):
+            response = self.client.post(
+                reverse('core:send'),
+                data                                = serialized_force_payment,
+                content_type                        = 'application/octet-stream',
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY      = self._get_encoded_provider_public_key(),
+                HTTP_CONCENT_OTHER_PARTY_PUBLIC_KEY = self._get_encoded_requestor_public_key(),
+            )
+
+        self.assertIsInstance(response, JsonResponse)
+        self.assertEqual(response.status_code, 503)
+
+    def test_in_shutdown_mode_gatekeeper_should_not_accept_download_tokens_from_clients(self):
+        """
+        Tests if in shutdown mode Gatekeeper will not accept new download tokens from clients.
+        """
+
+        upload_token                               = message.FileTransferToken()
```

Comment:
Did you mean `download_token`? ;)

Reply:
Yes, I was trying to do this too quickly. Will be fixed :+1: